src/util/DataResolver.js

```diff
@@ -80,42 +81,49 @@ class DataResolver {
    */
 
   /**
-   * Resolves a BufferResolvable to a Buffer.
+   * Resolves a BufferResolvable to a Buffer or a Stream.
    * @param {BufferResolvable|Stream} resource The buffer or stream resolvable to resolve
-   * @returns {Promise<Buffer>}
+   * @returns {Promise<Buffer|Stream>}
    */
   static resolveFile(resource) {
     if (!browser && Buffer.isBuffer(resource)) return Promise.resolve(resource);
     if (browser && resource instanceof ArrayBuffer) return Promise.resolve(Util.convertToBuffer(resource));
+    if (resource instanceof stream.Readable) return resource;
```

Comment:
```suggestion
    if (resource instanceof stream.Readable) return Promise.resolve(resource);
```
shouldn't Promise.resolve be used here too

Reply:
Just make the whole thing an async function and forget about it

Reply:
Yep, good catch!